wireshark_gen/templates/_ofclass_dissector.lua

```diff
@@ -43,7 +43,7 @@ function ${name}(reader, subtree)
 :: if ofclass.virtual:
     return ${ofclass.name}_v${version.wire_version}_dissectors[reader.peek(${ofclass.discriminator.offset},${ofclass.discriminator.length}):uint()](reader, subtree)
 :: else:
-:: if not ofclass.is_fixed_length:
+:: if not ofclass.is_fixed_length and ofclass.length_member is not None:
```

Comment:
Not sure of the context, but IIRC things like the packet payload in a
packet out are calculated implicitly by pkt_len = msg_len - header_size, so
there is no explicit length field for the payload packet.  This was at
least historically true.



On Tue, Feb 2, 2016 at 3:16 PM, Rich Lane <notifications@github.com> wrote:

> In wireshark_gen/templates/_ofclass_dissector.lua
> <https://github.com/floodlight/loxigen/pull/459#discussion_r51653073>:
>
> > @@ -43,7 +43,7 @@ function ${name}(reader, subtree)
> >  :: if ofclass.virtual:
> >      return ${ofclass.name}_v${version.wire_version}_dissectors[reader.peek(${ofclass.discriminator.offset},${ofclass.discriminator.length}):uint()](reader, subtree)
> >  :: else:
> > -:: if not ofclass.is_fixed_length:
> > +:: if not ofclass.is_fixed_length and ofclass.length_member is not None:
>
> Out of curiosity, what object is variable length and doesn't have a length
> member?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/floodlight/loxigen/pull/459/files#r51653073>.
>